calling-apis/problem-set-requests-with-js.md

```diff
@@ -0,0 +1,296 @@
+# Problem Set: Requests With JS
+
+## Directions
+
+Read the problem prompt, which will ask you to write JavaScript code to find certain answers.
+
+Once you found the answers, input them into the answers prompt below.
+
+## Preparation: LocationIQ Forward Geocoding API
+
+Use the [LocationIQ's Forward Geocoding API](https://locationiq.com/geocoding). Before starting this problem prompt:
+
+1. Obtain an API key for LocationIQ's Forward Geocoding API
+1. Familiarize yourself with the LocationIQ API Documentation
+1. Familiarize yourself with the `axios` library
+1. Open Postman for manual testing
+
+![Screenshot of LocationIQ's feature on the Geocoding API](../assets/calling-apis_problem-set-requests-with-js_location-iq-geocoding.png)
+
+## Problem Prompt
+
+Create a JavaScript program that finds the latitude and longitude coordinates for the [seven wonders of the New7Wonders of the World](https://en.wikipedia.org/wiki/New7Wonders_of_the_World), which are the following:
+
+| Wonder              |
+| ------------------- |
+| Great Wall of China |
+| Petra               |
+| Colosseum           |
+| Chichen Itza        |
+| Machu Picchu        |
+| Taj Mahal           |
+| Christ the Redeemer |
+
+### Extra Challenges
+
+See if you can accomplish these extra challenges. These challenges are tricky and interesting! One solution required extra research around the following subjects: `setTimeout`, "JavaScript sleep," `async` and `await`, and (traditional) `for` loops.
+
+1. Organize your code into functions and helper functions
+1. Use a loop
+1. Create this data structure to hold your answers
+
+```js
+{
+    "Great Wall of China": {
+        latitude: "...",
+        longitude: "..."
+    },
+    "Petra": {
+        latitude: "...",
+        longitude: "..."
+    },
+    "Colosseum": {
+        latitude: "...",
+        longitude: "..."
+    },
+    "Chichen Itza": {
+        latitude: "...",
+        longitude: "..."
+    },
+    "Machu Picchu": {
+        latitude: "...",
+        longitude: "..."
+    },
+    "Taj Mahal": {
+        latitude: "...",
+        longitude: "..."
+    },
+    "Christ the Redeemer": {
+        latitude: "...",
+        longitude: "..."
+    }
+}
+```
+
+### !callout-warning
+
+## Hint: Working with Rate Limits
+
+Rapidly making several HTTP requests to the same endpoint may exceed our API rate limit. There are many ways to work with the rate limit. As a suggestion, one strategy is to consider the following:
+
+1. How can we make requests more slowly?
+1. Research the `setTimeout` function, or how to implement a "sleep" function. We recommend making the program wait for 500 milliseconds to begin your experiment.
```

Comment:
rephrase attempting the pause logic so that it could be done in a separate program before bringing their findings into this solution
explicitly state that this is a good rate limiting delay

```suggestion
1. Research the `setTimeout` function, or how to implement a "sleep" function. We recommend trying to make a program wait for 500 milliseconds as a starting point. This is also a good pause to use between API calls!
```